### PR TITLE
[SINGULAR] DDP-8560: Dashboard Text and Graphic Updates

### PR DESCRIPTION
**Pull request by juanseba21**
1. Updated text image and button on the dashboard as per mockups (see https://broadinstitute.atlassian.net/browse/DDP-8560 )
2. Implemented a feature flag config which allows to switch any proper feature.
3. New modal appears when user clicks on "View Family Enrollment Message"

image

image

### CODE REVIEW

**Comment by alextanlan** [comment]
Can you please resolve the conflicts?

File: ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.html [hunk 1]
```diff
@@ -59,7 +59,19 @@ <h1 class="title">
                   <ng-container *ngTemplateOutlet="chevronRightSvg"></ng-container>
               </div>
           </button>
-          <p class="share-project" [innerHTML]="'ParticipantsList.ShareProject' | translate"></p>
+          <div class="family-enrollment-section" *ngIf="featureFlag_DDP_8506">
+            <img class="enrollment-image" src="/assets/images/step-1_DDP-8404.png" alt="enrollment image">
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.html]: Can you please add the alt translation?

File: ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.html [hunk 2]
```diff
@@ -59,7 +59,19 @@ <h1 class="title">
...
+            <div class="family-enrollment-description">
+              <h3 class="title-enrollment-message">Family Enrollment Message</h3>
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.html]: Can you please add the title translation?

File: ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.html [hunk 3]
```diff
@@ -138,5 +150,5 @@ <h1 class="title">
 
 <!--svg image container-->
 <ng-template #chevronRightSvg>
-    <img src="/assets/images/next.svg" alt="next button">
+  <img src="/assets/images/next.svg" alt="next button">
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.html]: Can you please add the alt translation?

File: ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.ts [hunk 4]
```diff
@@ -66,6 +70,7 @@ export class ParticipantsListComponent implements OnInit {
   ngOnInit(): void {
     this.clearParticipant();
     this.loadData();
+    console.log("#flag",this.featureFlag_DDP_8506);
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.ts]: Do we need `console.log` here? (also see lines 334, 379)

File: ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.ts [hunk 5]
```diff
@@ -403,7 +410,7 @@ export class ParticipantsListComponent implements OnInit {
 
   private hasOnlyAddParticipantActivity(activities: ActivityInstance[]): boolean {
     return (
-      activities.length === 1 &&
+      !!activities && activities.length === 1 &&
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.ts]: **#minor #optional**

`!!activities && activities.length === 1` 
can be shortened like
`activities?.length === 1`

File: ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.ts [hunk 6]
```diff
@@ -413,13 +420,23 @@ export class ParticipantsListComponent implements OnInit {
   }
 
   private filterOutAddParticipant(activities: ActivityInstance[]): ActivityInstance[] {
-    return activities.filter(
+    return !!activities ? activities.filter(
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.ts]: **#minor #optional**

`return !!activities ? activities.filter(...) : [];`
can be shortened like
`return (activities || []).filter(...)`

or just set a default value for the method argument, like

```
private filterOutAddParticipant(activities: ActivityInstance[] = []): ActivityInstance[] {
   return activities.filter(...);
}
```

File: ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.ts [hunk 7]
```diff
@@ -413,13 +420,23 @@ export class ParticipantsListComponent implements OnInit {
...
         ].includes(activity.activityCode as ActivityCode),
-    );
+    ): [];
   }
+
+  public openDisclaimerDialog(): void {
+    this.dialog.open(FamilyEnrollmentMessageComponent, {
+        width: '740px',
+        position: { top: '30px' },
+        data: {},
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.ts]: The `data` property can be skipped.

**Comment by alextanlan** [comment]
Can you please fix [linter warnings](https://app.circleci.com/pipelines/github/broadinstitute/ddp-angular/19106/workflows/e5d286bd-a846-4e5a-813f-d7ecfaee368b/jobs/28608) ?

File: ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.ts (hunk 6 above)
Comment (juanseba21) [reply ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.ts]: Done

File: ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.ts (hunk 5 above)
Comment (juanseba21) [reply ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.ts]: done

File: ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.ts (hunk 7 above)
Comment (juanseba21) [reply ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.ts]: done

File: ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.ts (hunk 4 above)
Comment (juanseba21) [reply ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.ts]: not really, good catch

File: ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.html (hunk 3 above)
Comment (juanseba21) [reply ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.html]: done

File: ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.html (hunk 2 above)
Comment (juanseba21) [reply ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.html]: done

File: ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.html (hunk 1 above)
Comment (juanseba21) [reply ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.html]: done

**Review by alextanlan — APPROVED** [review]

**Pull request merged**